python/pyspark/sql/connect/client/core.py

```diff
@@ -354,6 +355,22 @@ def get(self, key: str) -> Any:
         """
         return self.params[key]
 
+    @property
+    def session_id(self) -> Optional[str]:
+        """
+        Returns
+        -------
+        The session_id extracted from the parameters of the connection string or `None` if not
+        specified.
+        """
+        session_id = self.params.get(ChannelBuilder.PARAM_SESSION_ID, None)
+        if session_id is not None:
+            try:
+                uuid.UUID(session_id, version=4)
+            except ValueError as ve:
+                raise ValueError("Could not parse 'session_id' parameter for connection string", ve)
```

Comment:
same message as scala?
`"Parameter value 'session_id' must be a valid UUID format."`?

Reply:
It is done in test_connect_basic.py test, but actually not here, so that test will fail.